http-gateway/web/src/containers/things/websockets.js

```diff
@@ -43,7 +66,100 @@ export const deviceStatusListener = async message => {
         }
       })
     } catch (error) {} // ignore error
-  }, notificationsEnabled ? 2000 : 0)
+
+    // If the event was registered or unregistered, emit an event with the number to increment by
+    if ([REGISTERED, UNREGISTERED].includes(status)) {
+      // Emit an event: things-registered-unregistered-count
+      Emitter.emit(
+        THINGS_REGISTERED_UNREGISTERED_COUNT_EVENT_KEY,
+        deviceIds.length
+      )
+    }
+  }, notificationsEnabled ? DEFAULT_NOTIFICATION_DELAY : 0)
+}
+
+export const deviceResourceRegistrationListener = deviceId => message => {
+  // Things notifications must be enabled to see a toast message
+  const notificationsEnabled = isNotificationActive(
+    getThingNotificationKey(deviceId)
+  )(store.getState())
+
+  const data = JSON.parse(message.data)

```

Comment:
would rename this to parsedMessageData for clarity.

Reply:
Changed.